### ISSUE TITLE
build fails even after adding the -std=c99 flag

### ISSUE THREAD

**Issue by kolmai** [VirusTotal/yara-python#34]
At first the error suggests to use option -std=c99 in order to compile.

```
~/Downloads/yara-python$ python setup.py build
running build
running build_ext
building 'yara' extension
creating build
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/yara
creating build/temp.linux-x86_64-2.7/yara/libyara
creating build/temp.linux-x86_64-2.7/yara/libyara/modules
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara-python.c -o build/temp.linux-x86_64-2.7/yara-python.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/object.c -o build/temp.linux-x86_64-2.7/yara/libyara/object.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/lexer.c -o build/temp.linux-x86_64-2.7/yara/libyara/lexer.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/grammar.c -o build/temp.linux-x86_64-2.7/yara/libyara/grammar.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/sizedstr.c -o build/temp.linux-x86_64-2.7/yara/libyara/sizedstr.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/rules.c -o build/temp.linux-x86_64-2.7/yara/libyara/rules.o
yara/libyara/rules.c: In function ‘yr_rules_save_stream’:
yara/libyara/rules.c:743:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < YR_BITARRAY_NCHARS(MAX_THREADS); ++i) {
   ^
yara/libyara/rules.c:743:3: note: use option -std=c99 or -std=gnu99 to compile your code
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```
++++++++++++++++++++++++++++++++++++++++++++++++++++++++
After adding the required flag build still fails at sigjmp_buf function.
(unknown type name ‘sigjmp_buf’)
Please help how to proceed with the compilation process.
Thanks ahead!
++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```
~/Downloads/yara-python$ python setup.py build
running build
running build_ext
building 'yara' extension
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara-python.c -o build/temp.linux-x86_64-2.7/yara-python.o --std=c99
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/object.c -o build/temp.linux-x86_64-2.7/yara/libyara/object.o --std=c99
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/lexer.c -o build/temp.linux-x86_64-2.7/yara/libyara/lexer.o --std=c99
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/grammar.c -o build/temp.linux-x86_64-2.7/yara/libyara/grammar.o --std=c99
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/sizedstr.c -o build/temp.linux-x86_64-2.7/yara/libyara/sizedstr.o --std=c99
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/rules.c -o build/temp.linux-x86_64-2.7/yara/libyara/rules.o --std=c99
In file included from yara/libyara/rules.c:51:0:
yara/libyara/exception.h:81:1: error: unknown type name ‘sigjmp_buf’
 sigjmp_buf *exc_jmp_buf[MAX_THREADS];
 ^
yara/libyara/exception.h: In function ‘exception_handler’:
yara/libyara/exception.h:89:7: warning: passing argument 1 of ‘siglongjmp’ makes pointer from integer without a cast [enabled by default]
       siglongjmp(*exc_jmp_buf[tidx], 1);
       ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/assert.h:35,
                 from yara/libyara/rules.c:30:
/usr/include/x86_64-linux-gnu/bits/setjmp2.h:31:13: note: expected ‘struct __jmp_buf_tag *’ but argument is of type ‘int’
 extern void __REDIRECT_NTHNL (siglongjmp,
             ^
In file included from yara/libyara/rules.c:51:0:
yara/libyara/rules.c: In function ‘yr_rules_scan_mem_blocks’:
yara/libyara/exception.h:100:22: error: storage size of ‘oldact’ isn’t known
     struct sigaction oldact;                                    \
                      ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:101:22: error: storage size of ‘act’ isn’t known
     struct sigaction act;                                       \
                      ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:102:5: error: unknown type name ‘sigset_t’
     sigset_t oldmask;                                           \
     ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:101:12: warning: implicit declaration of function ‘sigemptyset’ [-Wimplicit-function-declaration]
     struct sigaction act;                                       \
            ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:101:12: warning: implicit declaration of function ‘pthread_sigmask’ [-Wimplicit-function-declaration]
     struct sigaction act;                                       \
            ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:108:21: error: ‘SIG_SETMASK’ undeclared (first use in this function)
     pthread_sigmask(SIG_SETMASK, &act.sa_mask, &oldmask);       \
                     ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:108:21: note: each undeclared identifier is reported only once for each function it appears in
     pthread_sigmask(SIG_SETMASK, &act.sa_mask, &oldmask);       \
                     ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:101:12: warning: implicit declaration of function ‘sigaction’ [-Wimplicit-function-declaration]
     struct sigaction act;                                       \
            ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:112:5: error: unknown type name ‘sigjmp_buf’
     sigjmp_buf jb;                                              \
     ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:101:12: warning: implicit declaration of function ‘sigsetjmp’ [-Wimplicit-function-declaration]
     struct sigaction act;                                       \
            ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:101:22: warning: unused variable ‘act’ [-Wunused-variable]
     struct sigaction act;                                       \
                      ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:100:22: warning: unused variable ‘oldact’ [-Wunused-variable]
     struct sigaction oldact;                                    \
                      ^
yara/libyara/rules.c:437:7: note: in expansion of macro ‘YR_TRYCATCH’
       YR_TRYCATCH({
       ^
yara/libyara/exception.h:100:22: error: storage size of ‘oldact’ isn’t known
     struct sigaction oldact;                                    \
                      ^
yara/libyara/rules.c:450:5: note: in expansion of macro ‘YR_TRYCATCH’
     YR_TRYCATCH({
     ^
yara/libyara/exception.h:101:22: error: storage size of ‘act’ isn’t known
     struct sigaction act;                                       \
                      ^
yara/libyara/rules.c:450:5: note: in expansion of macro ‘YR_TRYCATCH’
     YR_TRYCATCH({
     ^
yara/libyara/exception.h:102:5: error: unknown type name ‘sigset_t’
     sigset_t oldmask;                                           \
     ^
yara/libyara/rules.c:450:5: note: in expansion of macro ‘YR_TRYCATCH’
     YR_TRYCATCH({
     ^
yara/libyara/exception.h:112:5: error: unknown type name ‘sigjmp_buf’
     sigjmp_buf jb;                                              \
     ^
yara/libyara/rules.c:450:5: note: in expansion of macro ‘YR_TRYCATCH’
     YR_TRYCATCH({
     ^
yara/libyara/exception.h:101:22: warning: unused variable ‘act’ [-Wunused-variable]
     struct sigaction act;                                       \
                      ^
yara/libyara/rules.c:450:5: note: in expansion of macro ‘YR_TRYCATCH’
     YR_TRYCATCH({
     ^
yara/libyara/exception.h:100:22: warning: unused variable ‘oldact’ [-Wunused-variable]
     struct sigaction oldact;                                    \
                      ^
yara/libyara/rules.c:450:5: note: in expansion of macro ‘YR_TRYCATCH’
     YR_TRYCATCH({
     ^
yara/libyara/exception.h:100:22: error: storage size of ‘oldact’ isn’t known
     struct sigaction oldact;                                    \
                      ^
yara/libyara/rules.c:468:3: note: in expansion of macro ‘YR_TRYCATCH’
   YR_TRYCATCH({
   ^
yara/libyara/exception.h:101:22: error: storage size of ‘act’ isn’t known
     struct sigaction act;                                       \
                      ^
yara/libyara/rules.c:468:3: note: in expansion of macro ‘YR_TRYCATCH’
   YR_TRYCATCH({
   ^
yara/libyara/exception.h:102:5: error: unknown type name ‘sigset_t’
     sigset_t oldmask;                                           \
     ^
yara/libyara/rules.c:468:3: note: in expansion of macro ‘YR_TRYCATCH’
   YR_TRYCATCH({
   ^
yara/libyara/exception.h:112:5: error: unknown type name ‘sigjmp_buf’
     sigjmp_buf jb;                                              \
     ^
yara/libyara/rules.c:468:3: note: in expansion of macro ‘YR_TRYCATCH’
   YR_TRYCATCH({
   ^
yara/libyara/exception.h:101:22: warning: unused variable ‘act’ [-Wunused-variable]
     struct sigaction act;                                       \
                      ^
yara/libyara/rules.c:468:3: note: in expansion of macro ‘YR_TRYCATCH’
   YR_TRYCATCH({
   ^
yara/libyara/exception.h:100:22: warning: unused variable ‘oldact’ [-Wunused-variable]
     struct sigaction oldact;                                    \
                      ^
yara/libyara/rules.c:468:3: note: in expansion of macro ‘YR_TRYCATCH’
   YR_TRYCATCH({
   ^
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

```

**Comment by hex2a** [comment]
same problem (using debian jessie)

**Comment by nianelli** [comment]
I'm hitting the same error:

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.5 LTS"

Linux ubuntu 4.4.0-53-generic #74~14.04.1-Ubuntu SMP Fri Dec 2 03:43:31 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux


$ python setup.py build
running build
running build_ext
building 'yara' extension
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara-python.c -o build/temp.linux-x86_64-2.7/yara-python.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/compiler.c -o build/temp.linux-x86_64-2.7/yara/libyara/compiler.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/stream.c -o build/temp.linux-x86_64-2.7/yara/libyara/stream.o
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_MEMMEM=1 -DHASH_MODULE=1 -Iyara/libyara/include -Iyara/libyara/ -I. -I/usr/include/python2.7 -c yara/libyara/rules.c -o build/temp.linux-x86_64-2.7/yara/libyara/rules.o
yara/libyara/rules.c: In function ‘yr_rules_save_stream’:
yara/libyara/rules.c:743:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < YR_BITARRAY_NCHARS(MAX_THREADS); ++i) {
   ^
yara/libyara/rules.c:743:3: note: use option -std=c99 or -std=gnu99 to compile your code
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

**Comment by hnstance** [comment]
The issue missing c99/gnu99 standard was solved in 3.3 (https://github.com/VirusTotal/yara/issues/258), but I guess was broken again in 3.5.

**Comment by ni1-ni1** [comment]
Added the following to setup.py and it now works: extra_compile_args=['-std=gnu99'],

**Comment by wxsBSD** [comment]
Yeah, it looks like b3e80a1ba48063bb2f774416af906158230b1981 might have been the culprit here? Probably worth keeping slightly older toolchains around which we are likely to see in wide circulation?

**Comment by plusvic** [comment]
Fixed in ca3927fcebb85c429231087ef796753a37d5b079 and 228c2fa9a72080074a9f0bc41e1cb15ecc1180e5
